oracle-plugin/src/e2e-test/java/io.cdap.plugin/oracle/BQValidation.java

```diff
@@ -1,129 +1,146 @@
+  public static boolean compareResultSetData(ResultSet rsSource, List<JsonObject> bigQueryData) throws SQLException,
+    ParseException {
+    ResultSetMetaData mdSource = rsSource.getMetaData();
+    boolean result = false;
+    int columnCountSource = mdSource.getColumnCount();
 
-    return mapDifference.areEqual();
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
+    // Get the column count of the first JsonObject in bigQueryData
+    int columnCountTarget = 0;
+    if (bigQueryData.size() > 0) {
+      columnCountTarget = bigQueryData.get(0).entrySet().size();
+    }
+    // Compare the number of columns in the source and target
+    Assert.assertEquals("Number of columns in source and target are not equal",
+                        columnCountSource, columnCountTarget);
+
+    //Variable 'jsonObjectIdx' to track the index of the current JsonObject in the bigQueryData list,
+    int jsonObjectIdx = 0;
+    while (rsSource.next()) {
+      int currentColumnCount = 1;
+      while (currentColumnCount <= columnCountSource) {
+        String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+        int columnType = mdSource.getColumnType(currentColumnCount);
+        String columnName = mdSource.getColumnName(currentColumnCount);
+        // Perform different comparisons based on column type
+        switch (columnType) {
+          // Since we skip BFILE in Oracle Sink, we are not comparing the BFILE source and sink values
+          case OracleSourceSchemaReader.BFILE:
+            break;
+          case Types.BLOB:
+          case Types.VARBINARY:
+          case Types.LONGVARBINARY:
+            String sourceB64String = new String(Base64.getEncoder().encode(rsSource.getBytes(currentColumnCount)));
+            String targetB64String = bigQueryData.get(jsonObjectIdx).get(columnName).getAsString();
+            Assert.assertEquals("Different values found for column : %s",
+                                sourceB64String, targetB64String);
+            break;
+
+          case Types.NUMERIC:
+            long sourceVal = rsSource.getLong(currentColumnCount);
+            long targetVal = Long.parseLong(bigQueryData.get(jsonObjectIdx).get(columnName).getAsString());
+            Assert.assertTrue("Different values found for column : %s",
+                              String.valueOf(sourceVal).equals(String.valueOf(targetVal)));
+            break;
+
+          case Types.TIMESTAMP:
+            Timestamp sourceTS = rsSource.getTimestamp(columnName);
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss'Z'");
+            Date parsedDate = dateFormat.parse(bigQueryData.get(jsonObjectIdx).get(columnName).getAsString());
+            Timestamp targetTs = new java.sql.Timestamp(parsedDate.getTime());
+            result = sourceTS.equals(targetTs);
```

Comment:
Why are we using result variable in only in this case, any specific reason ?

Reply:
Previously it was just to check the boolean value while comparing the value of source and target.Now, I have changed it and comparing the sourceTS and targetTS values In Assert.assertEquals.